pkg/volume/flexvolume/flexvolume.go

```diff
@@ -241,76 +245,142 @@ func (f flexVolumeMounter) GetAttributes() volume.Attributes {
 // flexVolumeManager is the abstract interface to flex volume ops.
 type flexVolumeManager interface {
 	// Attaches the disk to the kubelet's host machine.
-	attach(mounter *flexVolumeMounter) (string, error)
+	attach(mounter *flexVolumeMounter, volOptions map[string]string) (string, error)
 	// Detaches the disk from the kubelet's host machine.
 	detach(unmounter *flexVolumeUnmounter, dir string) error
 	// Mounts the disk on the Kubelet's host machine.
-	mount(mounter *flexVolumeMounter, mnt, dir string) error
+	mount(mounter *flexVolumeMounter, mnt, dir string, volOptions map[string]string) error
 	// Unmounts the disk from the Kubelet's host machine.
-	unmount(unounter *flexVolumeUnmounter, dir string) error
+	unmount(unmounter *flexVolumeUnmounter, dir string) error
 }
 
 // SetUpAt creates new directory.
 func (f *flexVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
-
-	notmnt, err := f.blockDeviceMounter.IsLikelyNotMountPoint(dir)
-	if err != nil && !os.IsNotExist(err) {
-		glog.Errorf("Cannot validate mountpoint: %s", dir)
+	err := f.Attach(f.spec, "")
```

Comment:
Ummm what will be the difference between mount and mountdevice in the executable? In the current code, SetUpAt (minus Attach/WaitForAttach) only calls MountDevice with the result of GetDeviceMountPath. Then, MountDevice currently calls mount from the executable or falls back tooo default mount logic if mount is not supported by the executable.